dev/codeserver/java/com/google/gwt/dev/codeserver/dev_mode_on.js

```diff
@@ -122,19 +122,22 @@
     result.style.background = '#ddd';
     result.style.border = '2px outset #ddd';
     result.style.padding = '3pt';
-    result.setAttribute('href', 'javascript:' + encodeURIComponent(javascript));
+    result.onclick = javascript;
     result.title = 'Tip: drag this button to the bookmark bar';
```

Comment:
I don't think this will function as a bookmarklet at all if the href is replaced by a lambda - lambdas aren't copied when the bookmarklet URL's contents are... Is there a way we can detect that CSP is in use here and just don't offer the bookmarklet button?

Reply:
You are right. The button has a dual function (direct clicking or use as bookmarklet - I never used the second one, because it's server specific). Now I added both (CSP and bookmarklet) cases and it works for both cases (even if the browser shows a CSP issue). I could not think of a better way to separate the two cases better, while using only one button.